hamiltonian_generative_network.py

```diff
@@ -26,25 +26,31 @@ def __init__(self,
                  integrator,
                  optimizer,
                  loss,
+                 device,
+                 dtype,
                  seq_len,
                  channels=3):
         """Instantiate a Hamiltonian Generative Network.
 
```

Comment:
In line 152-153:

`        reconstruction_error_np = reconstruction_error.detach().cpu().numpy()
        kl_div_np = kl_div.detach().cpu().numpy() if kl_div is not None else None`
otherwise this blows up when using gpu.